src/composables/draw/draw-tooltip.ts

```diff
@@ -0,0 +1,129 @@
+import { Overlay } from 'ol'
+import { EventsKey, listen } from 'ol/events'
+import { Projection, transform } from 'ol/proj'
+import { getDistance as haversineDistance, getArea } from 'ol/sphere'
+import { unByKey } from 'ol/Observable'
+import { Circle, Geometry, LineString, Polygon } from 'ol/geom'
+import OlMap from 'ol/Map'
+import { DrawEvent } from 'ol/interaction/Draw'
+
+class DrawTooltip {
+  private measureTooltipElement: HTMLElement | null = null
+  private measureTooltipOverlay: Overlay | null = null
+  private changeEventKey: EventsKey | null = null
+
+  public add(map: OlMap, event: DrawEvent) {
+    this.createMeasureTooltip(map)
+    const geometry = event.feature.getGeometry()
+    console.assert(geometry !== undefined)
+    if (geometry === undefined) return
+    const proj = map.getView().getProjection()
+
+    this.changeEventKey = listen(
+      geometry,
+      'change',
+      () => this.updateTootip(geometry, proj),
+      this
+    )
+  }
+
+  public remove() {
+    this.removeMeasureTooltip()
+
+    if (this.changeEventKey !== null) {
+      unByKey(this.changeEventKey)
+      this.changeEventKey = null
+    }
+  }
+
+  private createMeasureTooltip(map: OlMap) {
+    this.removeMeasureTooltip()
+    this.measureTooltipElement = document.createElement('div')
+    this.measureTooltipElement.classList.add('lux-tooltip')
+
+    this.measureTooltipOverlay = new Overlay({
+      element: this.measureTooltipElement,
+      offset: [0, -15],
+      positioning: 'bottom-center',
+      stopEvent: false,
+    })
+    map.addOverlay(this.measureTooltipOverlay)
+  }
+
+  private removeMeasureTooltip() {
+    if (this.measureTooltipElement !== null) {
+      this.measureTooltipElement.parentNode?.removeChild(
+        this.measureTooltipElement
+      )
+      this.measureTooltipElement = null
+      this.measureTooltipOverlay = null
+    }
+  }
+
+  private updateTootip(geometry: Geometry, proj: Projection) {
+    let coord = undefined
+    let output = ''
+    if (geometry.getType() === 'LineString') {
```

Comment:
Are there const availaible in ol for geom type ? because I need this too in my PR, and I prefer to use const instead of string.

Reply:
Good point, I already had trouble importing the consts in my previous PR and just figured that they have [actually been removed in v6.15.0](https://github.com/openlayers/openlayers/commit/9a6f8493fb6ebe7829e39c5cd276b11e1a06c2c8) we use. See also https://openlayers.org/en/v6.15.0/apidoc/module-ol_geom_Geometry.html#~Type compared to https://openlayers.org/en/v6.5.0/apidoc/module-ol_geom_GeometryType.html.